src/GHC/DataSize.hs

```diff
@@ -56,12 +61,17 @@ closureSize x = do
 --
 --   A garbage collection is performed before the size is calculated, because
 --   the garbage collector would make heap walks difficult.
+--
+--   This function works very quickly on small data structures, but can be slow
+--   on large and complex ones. If speed is an issue its probably possible to
+--   get the exact size of a small portion of the data structure and then
+--   estimate the total size from that.
 
-recursiveSize :: Num b => a -> IO b
+recursiveSize :: a -> IO Word
 recursiveSize x = do
   performGC
   liftM snd $ go ([], 0) $ asBox x
-  where go (vs, acc) b@(Box y) = do
+  where go (!vs, !acc) b@(Box y) = do
```

Comment:
No I didn't, but I did wonder why it took about an hour to calculate a size that turned out to be a bit over 7 megabytes. When I get back to the office tomorrow I intend to compare this new version with the old.